### ISSUE TITLE
FEAT: Postgenerators

### ISSUE THREAD

**Issue by joszamama** [fandango-fuzzer/fandango#50]
At the moment we have pregenerators in grammars, (noted by :: generator at the grammar rule), but what about having post-generators? This could be extremely helpful in the case of TinyC, where we could generate files without taking into account if a variable has been declared previously (removing a complex constraint), and having a really quick and fast regex rule that for each variable (a=..), declare it first.

**Comment by andreas-zeller** [comment]
This should be addressed by #181. @joszamama, @alex9849 ?

**Comment by joszamama** [comment]
I dont think so, pregenerators with parameters are not postgenerators, right?

One could say that if you use nonterminal `<nT>` in a rule for nonterminal `<nT2>`, nT should be expanded, but this does not mean that nT2 is already expanded and we want to do some postprocessing.

**Comment by andreas-zeller** [comment]
Correct, but you can now post-process `<nT>` using a generator. For instance, you could have a generator `gen(<nT>)` that returns `<nT>`. However, it would also check the derivation tree `<nT>` for variables, and if some are undeclared, it would return `prefix + <nT>` where `prefix` is an appropriate declaration.
